mp2-v1/tests/common/cases/query/simple_select_queries.rs

```diff
@@ -0,0 +1,505 @@
+use futures::{stream, StreamExt, TryStreamExt};
+use itertools::Itertools;
+use log::info;
+use mp2_v1::{
+    api::MetadataHash,
+    indexing::{block::BlockPrimaryIndex, row::RowTreeKey, LagrangeNode},
+};
+use parsil::{
+    assembler::DynamicCircuitPis,
+    executor::{generate_query_execution_with_keys, generate_query_keys},
+    ParsilSettings, DEFAULT_MAX_BLOCK_PLACEHOLDER, DEFAULT_MIN_BLOCK_PLACEHOLDER,
+};
+use ryhope::{
+    storage::{pgsql::ToFromBytea, RoEpochKvStorage},
+    Epoch, NodePayload,
+};
+use sqlparser::ast::Query;
+use std::{fmt::Debug, hash::Hash};
+use tokio_postgres::Row as PgSqlRow;
+use verifiable_db::{
+    query::{
+        aggregation::{ChildPosition, NodeInfo},
+        computational_hash_ids::ColumnIDs,
+        universal_circuit::universal_circuit_inputs::{PlaceholderId, Placeholders},
+    },
+    revelation::{api::MatchingRow, RowPath},
+    test_utils::MAX_NUM_OUTPUTS,
+};
+
+use crate::common::{
+    cases::{
+        indexing::BLOCK_COLUMN_NAME,
+        planner::{IndexInfo, QueryPlanner, RowInfo, TreeInfo},
+        query::{
+            aggregated_queries::{
+                check_final_outputs, find_longest_lived_key, get_node_info, prove_single_row,
+            },
+            GlobalCircuitInput, QueryCircuitInput, RevelationCircuitInput, SqlReturn, SqlType,
+        },
+    },
+    proof_storage::{ProofKey, ProofStorage},
+    table::Table,
+    TestContext,
+};
+
+use super::QueryCooking;
+
+pub(crate) async fn prove_query<'a>(
+    mut parsed: Query,
+    table_hash: &MetadataHash,
+    planner: &mut QueryPlanner<'a>,
+    results: Vec<PgSqlRow>,
+) -> Result<()> {
+    let mut exec_query = generate_query_execution_with_keys(&mut parsed, &planner.settings)?;
+    let query_params = exec_query.convert_placeholders(&planner.query.placeholders);
+    let res = planner
+        .table
+        .execute_row_query(
+            &exec_query
+                .normalize_placeholder_names()
+                .to_pgsql_string_with_placeholder(),
+            &query_params,
+        )
+        .await?;
+    let matching_rows = res
+        .iter()
+        .map(|row| {
+            let key = RowTreeKey::from_bytea(row.try_get::<_, &[u8]>(0)?.to_vec());
+            let epoch = row.try_get::<_, Epoch>(1)?;
+            // all the other items are query results
+            let result = (2..row.len())
+                .filter_map(|i| {
+                    SqlType::Numeric.extract(&row, i).map(|res| match res {
+                        SqlReturn::Numeric(uint) => uint,
+                    })
+                })
+                .collect_vec();
+            Ok((key, epoch, result))
+        })
+        .collect::<Result<Vec<_>>>()?;
+    // compute input for each matching row
+    let row_tree_info = RowInfo {
+        satisfiying_rows: matching_rows
+            .iter()
+            .map(|(key, _, _)| key)
+            .cloned()
+            .collect(),
+        tree: &planner.table.row,
+    };
```

Comment:
I initially tried to do it with wide_lineage but I thought it was kind of useless because it will yield only the keys of the nodes, but then for each node I would still have to call `get_node_info` to build the path. So the code complexity didn't seem really worthy to me.

Reply:
:+1: